src/thing_creature.c

```diff
@@ -3433,9 +3432,16 @@ TbBool thing_is_creature_special_digger(const struct Thing *thing)
   return ((get_creature_model_flags(thing) & CMF_IsSpecDigger) != 0);
 }
 
-void anger_set_creature_anger_all_types(struct Thing *thing, long a2)
+void anger_set_creature_anger_all_types(struct Thing *thing, long new_value)
 {
-    _DK_anger_set_creature_anger_all_types(thing, a2);
+    // _DK_anger_set_creature_anger_all_types(thing, a2);
+    if (creature_can_get_angry(thing))
+    {
+      for (AnnoyMotive anger_type = AngR_NotPaid; anger_type < AngR_ListEnd; anger_type++ )
```

Comment:
starting to count at AngR_NotPaid really has no logical meaning here, if you were to say '1' instead you can just assume 0 means invalid and you start counting at the first real entry, saying you start counting at AngR_NotPaid  you can't assume that and have to look where it is in the list